frontend/src/Editor/Components/DropDown.jsx

```diff
@@ -109,9 +130,12 @@ export const DropDown = function DropDown({
   }, [label]);
 
   useEffect(() => {
-    setExposedVariable('optionLabels', display_values);
+    setExposedVariable(
+      'optionLabels',
+      schema?.filter((item) => item.visible)?.map((item) => item.label)
+    );
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [JSON.stringify(display_values)]);
+  }, [JSON.stringify(schema)]);
```

Comment:
We should have `display_values` and `advanced` as well. Whenever anything changes, we need expose `display_values` if `advanced` is false